### ISSUE TITLE
Unclear relation to `pnpm recursive install`

### ISSUE THREAD

**Issue by onetom** [pnpm/pnpmr#2]
Problem: https://pnpm.js.org/docs/usage/pnpm-cli.html#recursive suggests it implements the same functionality as `pnpmr`

Solution: Mention the relation to `pnpm recursive install` in the `README.md`. Provide an example scenario which demonstrates a typical use-case for `pnpmr`.

I'm trying to create a monorepo structure recommendation for interlinked node packages and node applications.

I found traces of documentation about the monorepo use-case in the `pnpm` project history:

https://github.com/pnpm/pnpm/commit/2d1e039874ef9f074c265c637b30c6ccca26b5a2#diff-c8f828151bef96c816fdad2b81ad25ae

but the [example it references](https://github.com/rstacruz/pnpm/examples/monorepo) is not available anymore, so I create my own example.

I tried multiple approaches:

1. Using `npx pnpmr` from the root of the monorepo: https://github.com/enumatech/pnpm-monorepo
2. With `pnpm recursive install`: https://github.com/enumatech/pnpm-monorepo/tree/builtin-recurse
3. Without a node package in the monorepo root: https://github.com/enumatech/pnpm-monorepo/tree/no-top-level-pkg

It would be good if we could agree on one of them as the most practical and advertise it in the `pnpm` documentation.

**Comment by zkochan** [comment]
You are right, there is `pnpm recursive install`. It currently does what `pnpmr` was doing but a lot more effectively. So I'll probably deprecate this package.

Currently `pnpm recursive install` is just finding all the projects in your monorepo and concurrently installs their dependencies in the correct order (dependencies first).

What you'd probably need is a command that would link the projects before installing their deps. I am planning to add a new command that will do exactly that `pnpm recursive link` (you can track, discuss it her https://github.com/pnpm/pnpm/issues/825)

**Comment by zkochan** [comment]
as for the examples, I will look into them more thoroughly later but I would vote for `pnpm recursive link` w/o `package.json` in the root of the repo.

The reason I don't like a `package.json` is because if it has dependencies, a node_modules will be created in the root of the repo. And that means that all the projects in the monorepo will have access to those dependencies. So a bug can be introduced, where a project uses a dependency that is not declared in its own `package.json`

**Comment by onetom** [comment]
I have the same gut feeling about having a `package.json` in the root of the monorepo. I can't think of a good use-case for it now. I only had such a setup before because that's how yarn workspaces are supposed to work.

So is it safe to say that what currently `pnpmr install` does will equivalent with the `pnpm recursive install; pnpm recursive link` sequence?

**Comment by zkochan** [comment]
`pnpm recursive install` already does what `pnpmr install` does. So pnpmr does not link packages. `pnpm recursive link` will link packages and install their deps. Kinda what `lerna bootstrap` does.

**Comment by zkochan** [comment]
@onetom [pnpm recursive link](https://pnpm.js.org/docs/usage/pnpm-cli.html#recursive-link) has been published wit [pnpm@1.32.0](https://github.com/pnpm/pnpm/releases/tag/v1.32.0)

**Comment by onetom** [comment]
Tested `pnpm recursive install & pnpm recursive link` with `v1.32.0` on [this repo](https://gitlab.com/oax/asset-gateway-poc/tree/d90256e409f9b68810e669b5948e64db19149cb6) and it works as expected.
We can close this issue probably and mark this tool deprecated.
I'm wondering though if it would make sense for `pnpm recursive install` to do a link too, since a `pnpm install` does linking automatically.

**Comment by zkochan** [comment]
What do you mean? `pnpm install` doesn't do linking automatically. It just doesn't override links that were previously created by `pnpm link`
